### ISSUE TITLE
R8 Compile error on 3.0.0

### ISSUE THREAD

**Issue by cruddell001** [getsentry/sentry-android-gradle-plugin#299]
_Android Gradle Plugin:_
**3.0.0**

_Gradle:_
**7.1.2**

_Optimizer:_
- [x]  ProGuard
- [ ]  R8

_Have you created the sentry.properties file?_
- [x]  Yes -> where? - project root folder
- [ ]  No

_SDKs:_
- [x]  sentry-android -> which version? - 5.7.0
- [ ]  sentry-react-native -> which version?

---
I have the following issue:
When trying to use version 3.0.0 of the gradle plugin, I am receiving a compile error of:
`com.android.tools.r8.internal.YI0: Different stack heights at jump target: 1 != 0`

Looking at the stack trace, it appears the plugin may be running into a conflict with these libraries we are using:
```
    implementation "com.google.android.gms:play-services-vision:20.1.3"
    implementation "com.google.android.gms:play-services-mlkit-text-recognition:18.0.0"
```

**Note: Using version 2.1.5 of the plugin does not result in the error.**

stack trace:
```
[sentry] Error while instrumenting ModelUtils.getModelLoggingInfo (Landroid/content/Context;Lcom/google/mlkit/common/model/LocalModel;)Lcom/google/mlkit/common/internal/model/ModelUtils$ModelLoggingInfo;.
Please report this issue at https://github.com/getsentry/sentry-android-gradle-plugin/issues
java.lang.ArrayIndexOutOfBoundsException
AGPBI: {"kind":"error","text":"com.android.tools.r8.internal.YI0: Different stack heights at jump target: 1 != 0","sources":[{"file":"/Users/chris.ruddell/.gradle/caches/transforms-3/15c8ad1a1ca054b045d1fd573acc4388/transformed/jetified-vision-common-16.5.0-runtime.jar"}],"tool":"D8"}

> Task :app:processProductionDebugResources

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:mergeExtDexProductionDebug'.
> Could not resolve all files for configuration ':app:productionDebugRuntimeClasspath'.
   > Failed to transform vision-common-16.5.0.aar (com.google.mlkit:vision-common:16.5.0) to match attributes {artifactType=android-dex, asm-transformed-variant=productionDebug, dexing-enable-desugaring=true, dexing-is-debuggable=true, dexing-min-sdk=24, org.gradle.status=release, org.gradle.usage=java-runtime}.
      > Execution failed for DexingNoClasspathTransform: /Users/chris.ruddell/.gradle/caches/transforms-3/15c8ad1a1ca054b045d1fd573acc4388/transformed/jetified-vision-common-16.5.0-runtime.jar.
         > Error while dexing.
   > Failed to transform common-17.1.1.aar (com.google.mlkit:common:17.1.1) to match attributes {artifactType=android-dex, asm-transformed-variant=productionDebug, dexing-enable-desugaring=true, dexing-is-debuggable=true, dexing-min-sdk=24, org.gradle.status=release, org.gradle.usage=java-runtime}.
      > Execution failed for AsmClassesTransform: /Users/chris.ruddell/.gradle/caches/transforms-3/d99cbf816166962de6fc99b391048d5d/transformed/jetified-common-17.1.1-runtime.jar.
         > java.lang.ArrayIndexOutOfBoundsException (no error message)
   > Failed to transform common-17.1.1.aar (com.google.mlkit:common:17.1.1) to match attributes {artifactType=android-dex, asm-transformed-variant=productionDebug, dexing-enable-desugaring=true, dexing-is-debuggable=true, dexing-min-sdk=24, org.gradle.status=release, org.gradle.usage=java-runtime}.
      > Execution failed for AsmClassesTransform: /Users/chris.ruddell/.gradle/caches/transforms-3/d99cbf816166962de6fc99b391048d5d/transformed/jetified-common-17.1.1-runtime.jar.
         > java.lang.ArrayIndexOutOfBoundsException (no error message)

* Try:
Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* ```

**Steps to reproduce:**
- Try to build and run the app

**Actual result:**
- Compile time error

**Expected result:**
- App runs

**Comment by marandaneto** [comment]
@cruddell001 Thanks for reporting, can you try to disable the `tracingInstrumentation` feature and see if that works?
https://docs.sentry.io/platforms/android/gradle/#configure

**Comment by PaulWoitaschek** [comment]
I can confirm that disabling tracingInstrumentation fixes this on agp 7.1.2, sentry 3.0.0 and gradle 7.4.1

**Comment by romtsn** [comment]
@PaulWoitaschek do you have the same libs causing this issue, or some other ones?

**Comment by PaulWoitaschek** [comment]
It's also mlkit

**Comment by romtsn** [comment]
thanks for your patience and sorry for the inconvenience. We just shipped `3.0.1` with a fix for this issue, please give it a try and let us know if there are still issues. Thanks!
